nuxeo-runtime/nuxeo-connect-standalone/src/test/java/org/nuxeo/launcher/connect/TestConnectBroker.java

```diff
@@ -284,12 +285,66 @@ public void testDownloadUnknownPackage() throws Exception {
 
         // Assertions
         checkPackagesState(connectBroker, Arrays.asList("unknown-package"), NON_EXISTENT);
+        connectBroker.getCommandSet().log();
         List<String> expectedLogs = new ArrayList<>();
         expectedLogs.add("Downloading [unknown-package]...");
         expectedLogs.add("\tDownload failed (not found).");
         checkLogEvents(expectedLogs, logCaptureResult.getCaughtEvents());
     }
 
+    /**
+     * Reproduction of the commands in <a href="https://jira.nuxeo.com/browse/NXP-21573">NXP-21573</a>, which should be
+     * successful:
+     *
+     * <pre><code>
+     * mp-add $TEST_LOCAL_ONLY_PATH/F-1.0.0-SNAPSHOT.zip
+     * mp-install F
+     * </code></pre>
+     */
+    @Test
+    @LogCaptureFeature.FilterWith(PkgRequestLogFilter.class)
+    public void testDownloadUnknownSnapshotPackage() throws Exception {
+        // Environment setup
+        Environment.getDefault().setProperty(Environment.DISTRIBUTION_NAME, "server");
+        Environment.getDefault().setProperty(Environment.DISTRIBUTION_VERSION, "8.3");
+        ConnectBroker connectBroker = new ConnectBroker(Environment.getDefault());
+        ((StandaloneCallbackHolder) NuxeoConnectClient.getCallBackHolder()).setTestMode(true);
+        connectBroker.setAllowSNAPSHOT(true);
+        final PackageState NON_EXISTENT = null;
+
+        // Given an unknown package 'F-1.0.0-SNAPSHOT', which is:
+        // - only available from local file system
+        // - not yet added into the Nuxeo Server 8.3
+        assertThat(new File(TEST_LOCAL_ONLY_PATH + "/F-1.0.0-SNAPSHOT.zip")).exists();
+        checkPackagesState(connectBroker, Arrays.asList("F-1.0.0-SNAPSHOT"), NON_EXISTENT);
+
+        // When enable the relax mode, and mp-add $TEST_LOCAL_ONLY_PATH/F-1.0.0-SNAPSHOT.zip
+        connectBroker.setRelax("true");
```

Comment:
Actually we don't need this for this command: adding a local ZIP file does not require any relax mode control.

However, this line is still needed for the 2nd command `mp-install F`, because installing a local package which is not available in remote, requires a relax mode confirmation:

```
Package(s) F not available on platform version server-8.3.
Do you want to relax the constraint (yes/no)? [no] yes
```

This line is now moved to the 2nd command.